homeassistant/components/blue_current/config_flow.py

```diff
@@ -52,10 +57,26 @@ async def async_step_user(
 
             else:
                 await self.async_set_unique_id(customer_id)
-                self._abort_if_unique_id_configured()
 
+                if self._reauth_entry:
+                    self.hass.config_entries.async_update_entry(
+                        self._reauth_entry, data=user_input
+                    )
+                    await self.hass.config_entries.async_reload(
+                        self._reauth_entry.entry_id
+                    )
+                    return self.async_abort(reason="reauth_successful")
+
+                self._abort_if_unique_id_configured()
```

Comment:
Instead of calling `await self.async_set_unique_id` above, let's only call that when there is no `self._reauth_entry`. We should also check if the unique_id we just got is the same as the one in the reauth entry.

I think the flow in the youtube integration would be a good example here.